src/IPNFT.sol

```diff
@@ -140,6 +141,34 @@ contract IPNFT is ERC721URIStorageUpgradeable, ERC721BurnableUpgradeable, IReser
         return reservationId;
     }
 
+    function mintReservation(
```

Comment:
replace the other function with this one. There are no mintpasses after this change.